Lib/test/test_dataclasses.py

```diff
@@ -3024,6 +3024,16 @@ class C:
 
         replace(c, x=5)
 
+    def test_initvar_is_specified(self):
+        @dataclass
+        class C:
+            x: int
+            y: InitVar[int]
+
+        c = C(1, 10)
+        with self.assertRaisesRegex(ValueError, r"InitVar 'y' must be "
+                                    "specified with replace()"):
+            replace(c, i=3)
```

Comment:
Please add a test that shows that an `InitVar` does work with `replace()`, and check that its value is used (probably via a `__post_init__` method). Maybe something like:
```
def __post_init__(self, y):
   self.x *= y
```
or similar.